### PR TITLE
Resolved scim failing due to missing rfc cases

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Description 📣

This PR address our failing scim operations due to missing various corner cases in scim operation. This adds all the proper scim conventions as needed

## Type ✨

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [ ] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/server/app.ts [hunk 1]
```diff
@@ -49,6 +49,21 @@ export const main = async ({ db, smtp, logger, queue, keyStore }: TMain) => {
...
+        done(null, undefined);
+        return;
+      }
+      const json: unknown = JSON.parse(strBody);
+      done(null, json);
+    } catch (err) {
+      const error = err as Error;
+      done(error, undefined);
+    }
+  });
```
Comment (sheensantoscapadngan) [inline backend/src/server/app.ts:65]: wait why did we have to move this out of the SCIM router scope?

Comment (akhilmhdh) [reply backend/src/server/app.ts:65]: It doesn't matter. At the end of the day scim json is normal json.

File: backend/src/lib/knex/scim.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        void query.whereNot((subQueryBuilder) => {
+          stack.push({
+            query: subQueryBuilder,
+            scimFilterAst: appendParentToGroupingOperator(scimFilterAst.attrPath, scimFilterAst.valFilter)
+          });
+        });
+        break;
+      }
+      default:
+        break;
```
Comment (sheensantoscapadngan) [inline backend/src/lib/knex/scim.ts:118]: which part is this? 
image

**Review by sheensantoscapadngan — APPROVED** [review]

**Review by dangtony98 — APPROVED** [review]
LGTM
